src/containers/TexterTodo.jsx

```diff
@@ -7,26 +7,46 @@ import gql from 'graphql-tag'
 class TexterTodo extends React.Component {
   componentWillMount() {
     const { assignment } = this.props.data
+    this.assignContactsIfNeeded()
     if (!assignment || assignment.campaign.isArchived) {
       this.props.router.push(
         `/app/${this.props.params.organizationId}/todos`
       )
     }
   }
 
-  refreshAssignmentContacts = () => this.props.data.refetch()
+  assignContactsIfNeeded = async (checkServer = false) => {
+    const { assignment } = this.props.data
+    if ((assignment.contacts.length == 0 || checkServer) && assignment.campaign.useDynamicAssignment) {
+      
+      const didAddContacts = await this.props.mutations.findNewCampaignContact(assignment.id, 1)
+
+      if (didAddContacts){
+        this.props.data.refetch()
+      } else {
+        this.props.router.push(
```

Comment:
This looks like an awesome feature (that I believe Color of Change is looking for as well) for auto-refilling contact assignments.
I only have nits:
* don't include a count from the client in terms of how many to get -- let the server do that (that way it can be a config option, rate-limited, etc)
* maybe keep the `props.router.push()` call in a separate method which is either called here, or sent as an argument to `assignContactsIfNeeded` (that way getting new assignments can be orthogonal to the 'very end')

Reply:
this is awesome!